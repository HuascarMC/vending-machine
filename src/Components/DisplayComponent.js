import React from 'react';

class DisplayComponent extends React.Component {
  render() {
    return(
      <div className="display-wrapper">
        <ul>
          <li><p className="price">$0.65<button className="coke display-option" onClick={() => this.props.updateItem("coke") } >COKE</button></p></li>
          <li><p className="price">$1.00<button className="pepsi display-option" onClick={() => this.props.updateItem("pepsi") }>PEPSI</button></p></li>
          <li><p className="price">$1.50<button className="soda display-option" onClick={() => this.props.updateItem("soda") }>SODA</button></p></li>
          <li><p className="price">$2.00<button className="water display-option" onClick={() =>this.props.updateItem("water") }>WATER</button></p></li>
        </ul>
      </div>
    );
  }
}

export default DisplayComponent;
