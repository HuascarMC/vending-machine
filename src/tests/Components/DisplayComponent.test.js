import React from 'react';
import DisplayComponent from '../../Components/DisplayComponent'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('should render without crashing', () => {
  shallow(<DisplayComponent value={0.00}/>);
});

it('should render machine-display div', () => {
  const wrapper = shallow(<DisplayComponent value={0.00}/>);
  const div = wrapper.find('.machine-display');
  expect(div.exists()).toEqual(true);
});
