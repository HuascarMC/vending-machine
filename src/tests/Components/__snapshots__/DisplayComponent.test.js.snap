// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="display machine-display"
>
  <p
    className="display-text"
  >
    0
  </p>
</div>
`;
